RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate subscriptions/rooms to TS

## Proposed changes
Migrate subscriptions/rooms to TS.

## Issue(s)	
N/A

## How to test or reproduce
Open a chat: 
* check if the messages are loading;
* check if the messages are updated; 
* check the threads;
* check if leave a room is still working;

## Screenshots
N/A

## Types of changes




## Checklist



## Further comments
